File: cpp/Platform.Data.Doublets.Sequences.Tests/UnaryNumberConvertersTests.cpp
namespace Platform::Data::Doublets::Sequences::Tests
{
    TEST_CLASS(UnaryNumberConvertersTests)
    {
        public: TEST_METHOD(ConvertersTest)
        {
            using (auto scope = TempLinksTestScope())
            {
                inline static const std::int32_t N = 10;
                auto links = scope.Links;
                auto meaningRoot = links.CreatePoint();
                auto one = links.CreateAndUpdate(meaningRoot, links.Constants.Itself);
                auto powerOf2ToUnaryNumberConverter = PowerOf2ToUnaryNumberConverter<std::uint64_t>(links, one);
                auto toUnaryNumberConverter = AddressToUnaryNumberConverter<std::uint64_t>(links, powerOf2ToUnaryNumberConverter);
                std::srand(0);
                std::uint64_t numbers[] = std::uint64_t[N];
                std::uint64_t unaryNumbers[] = std::uint64_t[N];
                for (std::int32_t i = 0; i < N; i++)
                {
                    numbers[i] = random.NextUInt64();
                    unaryNumbers[i] = toUnaryNumberConverter.Convert(numbers[i]);
                }
                auto fromUnaryNumberConverterUsingOrOperation = UnaryNumberToAddressOrOperationConverter<std::uint64_t>(links, powerOf2ToUnaryNumberConverter);
                auto fromUnaryNumberConverterUsingAddOperation = UnaryNumberToAddressAddOperationConverter<std::uint64_t>(links, one);
                for (std::int32_t i = 0; i < N; i++)
                {
                    Assert::AreEqual(numbers[i], fromUnaryNumberConverterUsingOrOperation.Convert(unaryNumbers[i]));
                    Assert::AreEqual(numbers[i], fromUnaryNumberConverterUsingAddOperation.Convert(unaryNumbers[i]));
                }
            }
        }
    };
}

File: csharp/Platform.Data.Doublets.Sequences.Tests/UnaryNumberConvertersTests.cs
// using Xunit;
// using Platform.Random;
// using Platform.Data.Doublets.Numbers.Unary;
//
// namespace Platform.Data.Doublets.Sequences.Tests
// {
//     public static class UnaryNumberConvertersTests
//     {
//         [Fact]
//         public static void ConvertersTest()
//         {
//             using (var scope = new TempLinksTestScope())
//             {
//                 const int N = 10;
//                 var links = scope.Links;
//                 var meaningRoot = links.CreatePoint();
//                 var powerOf2ToUnaryNumberConverter = new PowerOf2ToUnaryNumberConverter<ulong>(links, TLinkAddress.One);
//                 var toUnaryNumberConverter = new AddressToUnaryNumberConverter<ulong>(links, powerOf2ToUnaryNumberConverter);
//                 var random = new System.Random(0);
//                 ulong[] numbers = new ulong[N];
//                 ulong[] unaryNumbers = new ulong[N];
//                 for (int i = 0; i < N; i++)
//                 {
//                     numbers[i] = random.NextUInt64();
//                     unaryNumbers[i] = toUnaryNumberConverter.Convert(numbers[i]);
//                 }
//                 var fromUnaryNumberConverterUsingOrOperation = new UnaryNumberToAddressOrOperationConverter<ulong>(links, powerOf2ToUnaryNumberConverter);
//                 var fromUnaryNumberConverterUsingAddOperation = new UnaryNumberToAddressAddOperationConverter<ulong>(links, TLinkAddress.One);
//                 for (int i = 0; i < N; i++)
//                 {
//                     Assert.Equal(numbers[i], fromUnaryNumberConverterUsingOrOperation.Convert(unaryNumbers[i]));
//                     Assert.Equal(numbers[i], fromUnaryNumberConverterUsingAddOperation.Convert(unaryNumbers[i]));
//                 }
//             }
//         }
//     }
// }
